src/Oauth1.php

```diff
@@ -239,8 +239,11 @@ private function prepareParameters($data)
      */
     private function signUsingHmacSha1($baseString)
     {
-        $key = rawurlencode($this->config['consumer_secret'])
-            . '&' . rawurlencode($this->config['token_secret']);
+        $key = rawurlencode($this->config['consumer_secret']);
+
+        if (!empty($this->config['token_secret'])) {
+            $key .= '&' . rawurlencode($this->config['token_secret']);
```

Comment:
The `&` character should be left outside the `if` structure, otherwise the key format won't respect the [section 9.1.3](http://oauth.net/core/1.0a/#rfc.section.9.1.3) of the RFC.